bundles/http/Citrix/Kronos-Workforce-Central/f3b61af4-efd0-43c2-9a08-a467fc8242c1.js

```diff
@@ -0,0 +1,948 @@
+        }
+        return {
+            "endDate":moment(value.$.EndDate,integrationParameters.dateFormat).format('yyyy-MM-DD'),
+            "endTime":value.$.EndTime,
+            "startDate":moment(value.$.StartDate,integrationParameters.dateFormat).format('yyyy-MM-DD'),
+            "startTime":value.$.StartTime,
+            "personNumber":person.personNumber,
+            "StatusName":value.$.StatusName,
+            "Length":value.$.Length,
+            "requestId":value.$.Id,
+            "CommentText":com.$?.CommentText ?? null,
+            "CommentDataSourceClientName":com.Notes?.[0].Note[0].DataSource[0].DataSource[0].$.ClientName ?? null,
+            "CommentDataSourceFunctionalAreaCode":com.Notes !=undefined ?Number(com.Notes[0].Note[0].DataSource[0].DataSource[0].$.FunctionalAreaCode):null,
+            "CommentDataSourceFunctionalAreaName":com.Notes?.[0].Note[0].DataSource[0].DataSource[0].$.FunctionalAreaName ?? null,
+            "CommentDataSourceServerName":com.Notes?.[0].Note[0].DataSource[0].DataSource[0].$.ServerName ?? null,
+            "CommentDataSourceUserName":com.Notes?.[0].Note[0].DataSource[0].DataSource[0].$.UserName ?? null,
+            "CommentNotesText":com.Notes?.[0].Note[0].$.Text ?? null,
+            "CommentNotesTimestamp":com.Notes?.[0].Note[0].$.Timestamp ?? null,
+            "CreationDateTime":moment(value.$.CreationDateTime, `${integrationParameters.dateFormat} hh:mm A`).toDate(),
+        }
+    })
+}
+
+
+
+function getValidDateFormat(dateFormat){
+    let validFormats = new Set(['YYYY/MM/DD', 'MM/DD/YYYY', 'YYYY-MM-DD', 'MM-DD-YYYY','M-DD-YYYY', 'M/DD/YYYY', 'DD-MM-YYYY'])
+    if(!validFormats.has(dateFormat.toUpperCase())){
+        throw new Error(`Date Format [${dateFormat}] is not supported , choose any of the supported formats [${[...validFormats].toString()}] `)
+    }
+    return dateFormat.toUpperCase()
+}
+
+async function incrementalSync({ dataStore, client,latestSynchronizationTime, integrationParameters,context}){
+    integrationParameters.dateFormat = getValidDateFormat(integrationParameters.dateFormat)
+    const session = await logIn(client, integrationParameters.username , integrationParameters.password);
+    const request = createRequest(client,session)
+    await Promise.all([
+        syncRequestTimeOff(dataStore, request, context.people,integrationParameters,latestSynchronizationTime),
+        syncworkTime(dataStore, request, context.people,integrationParameters,latestSynchronizationTime)
+    ])
+}
+async function fullSync({ dataStore, client, integrationParameters,context}) {
+    integrationParameters.dateFormat = getValidDateFormat(integrationParameters.dateFormat)
+    const session = await logIn(client, integrationParameters.username , integrationParameters.password);
+    const people = await syncPeople(dataStore, client, session,integrationParameters);
+    context.people = people
```

Comment:
Good idea to use context here.
But it would be awesome if you store only array of `personNumber`s because you do not need any other attributes from a person.